go/response_handlers.go

```diff
@@ -215,6 +235,100 @@ func parseSet(response *C.struct_CommandResponse) (any, error) {
 	return slice, nil
 }
 
+// parseLCSMatchedPositions converts the nested array structure from LCSWithOptions into a slice of LCSMatchedPosition structs
+// The input structure has the shape of:
+// ```
+// 1) 1) 1) (integer) 4
+//  2. (integer) 7
+//  2. 1) (integer) 5
+//  2. (integer) 8
+//  2. 1) 1) (integer) 2
+//  2. (integer) 3
+//  2. 1) (integer) 0
+//  2. (integer) 1
+//
+// ```
+// which represents matched positions between two strings
+func parseLCSMatchedPositions(matches any) ([]models.LCSMatchedPosition, error) {
+	if matches == nil {
+		return []models.LCSMatchedPosition{}, nil
+	}
+
+	matchesArray, ok := matches.([]any)
+	if !ok {
+		return nil, fmt.Errorf("expected matches to be an array, got %T", matches)
+	}
+
+	result := make([]models.LCSMatchedPosition, len(matchesArray))
+	for i, match := range matchesArray {
+		matchArray, ok := match.([]any)
+		if !ok {
+			return nil, fmt.Errorf("expected match to be an array, got %T ", match)
+		}
+
+		if len(matchArray) != 2 && len(matchArray) != 3 {
+			return nil, fmt.Errorf("expected match to be an array of length 2 or 3, got %T with length %d", matchArray, len(matchArray))
+		}
+
+		// Parse Key1 position
+		key1Array, ok := matchArray[0].([]any)
+		if !ok || len(key1Array) != 2 {
+			return nil, fmt.Errorf("expected key1 to be an array of length 2, got %T with length %d", matchArray[0], len(key1Array))
+		}
+
+		key1Start, err := convertToInt64(key1Array[0])
+		if err != nil {
+			return nil, fmt.Errorf("expected key1 start to be a number, got %T", key1Array[0])
+		}
+
+		key1End, err := convertToInt64(key1Array[1])
+		if err != nil {
+			return nil, fmt.Errorf("expected key1 end to be a number, got %T", key1Array[1])
+		}
+
+		// Parse Key2 position
+		key2Array, ok := matchArray[1].([]any)
+		if !ok || len(key2Array) != 2 {
+			return nil, fmt.Errorf("expected key2 to be an array of length 2, got %T with length %d", matchArray[1], len(key2Array))
+		}
+
+		key2Start, err := convertToInt64(key2Array[0])
+		if err != nil {
+			return nil, fmt.Errorf("expected key2 start to be a number, got %T", key2Array[0])
+		}
+
+		key2End, err := convertToInt64(key2Array[1])
+		if err != nil {
+			return nil, fmt.Errorf("expected key2 end to be a number, got %T", key2Array[1])
+		}
+
+		var matchLen int64
+		if len(matchArray) == 2 {
+			// Calculate match length (end - start + 1)
+			matchLen = key1End - key1Start + 1
```

Comment:
We should calc nothing.
Maybe we should have different structs for different APIs.

Reply:
You can change this to `Result[int]` then